pkg/resourceinterpreter/configurableinterpreter/configmanager/manager.go

```diff
@@ -0,0 +1,107 @@
+type ConfigManager interface {
+	LuaScriptAccessors() map[schema.GroupVersionKind]LuaScriptAccessor
+	HasSynced() bool
+}
+
+// interpreterConfigManager  collect the resource interpreter customization configuration.
+type interpreterConfigManager struct {
+	initialSynced *atomic.Value
+	lister        cache.GenericLister
+	configuration *atomic.Value
+}
+
+// LuaScriptAccessors return all configured resource by lua.
+func (configManager *interpreterConfigManager) LuaScriptAccessors() map[schema.GroupVersionKind]LuaScriptAccessor {
+	return configManager.configuration.Load().(map[schema.GroupVersionKind]LuaScriptAccessor)
+}
+
+// HasSynced return true when the manager is synced with existing configuration.
+func (configManager *interpreterConfigManager) HasSynced() bool {
+	if configManager.initialSynced.Load().(bool) {
+		return true
+	}
+
+	configuration, err := configManager.lister.List(labels.Everything())
+	klog.Infof("interpreterConfigManager HasSynced %v", configuration)
+	if err == nil && len(configuration) == 0 {
+		// the empty list we initially stored is valid to use.
+		// Setting initialSynced to true, so subsequent checks
+		// would be able to take the fast path on the atomic boolean in a
+		// cluster without any resource interpreter customizations configured.
+		configManager.initialSynced.Store(true)
+		// the informer has synced, and we don't have any items
+		return true
+	}
+	return false
+}
+
+// NewInterpreterConfigManager return a new interpreterConfigManager with resource interpreter customizations handlers.
+func NewInterpreterConfigManager(inform genericmanager.SingleClusterInformerManager) ConfigManager {
+	manager := &interpreterConfigManager{
+		lister:        inform.Lister(resourceInterpreterCustomizationsGVR),
+		initialSynced: &atomic.Value{},
+		configuration: &atomic.Value{},
+	}
+	manager.configuration.Store(make(map[schema.GroupVersionKind]LuaScriptAccessor))
+	manager.initialSynced.Store(false)
+	configHandlers := fedinformer.NewHandlerOnEvents(
+		func(_ interface{}) { manager.updateConfiguration() },
+		func(_, _ interface{}) { manager.updateConfiguration() },
+		func(_ interface{}) { manager.updateConfiguration() })
+	inform.ForResource(resourceInterpreterCustomizationsGVR, configHandlers)
+	return manager
+}
+
+func (configManager *interpreterConfigManager) updateConfiguration() {
+	configurations, err := configManager.lister.List(labels.Everything())
+	if err != nil {
+		utilruntime.HandleError(fmt.Errorf("error updating configuration: %v", err))
+		return
+	}
+	configs := make(map[schema.GroupVersionKind]CustomAccessor, 0)
+
+	for _, c := range configurations {
+		unstructuredConfig, err := helper.ToUnstructured(c)
+		if err != nil {
+			klog.Errorf("Failed to transform ResourceInterpreterCustomization: %w", err)
+			return
+		}
+		config := &configv1alpha1.ResourceInterpreterCustomization{}
+		err = helper.ConvertToTypedObject(unstructuredConfig, config)
```

Comment:
I don't get why convert object to `unstructured`, then to `typed`.

Reply:
Modify like this?
```go
-               unstructuredConfig, err := helper.ToUnstructured(c)
-               if err != nil {
-                       klog.Errorf("Failed to transform ResourceInterpreterCustomization: %w", err)
-                       return
-               }
-               config := &configv1alpha1.ResourceInterpreterCustomization{}
-               err = helper.ConvertToTypedObject(unstructuredConfig, config)
-               if err != nil {
-                       gvk := unstructuredConfig.GroupVersionKind().String()
-                       klog.Errorf("Failed to convert object(%s), err: %v", gvk, err)
-                       return
-               }
+               config := c.(*configv1alpha1.ResourceInterpreterCustomization)

```

Reply:
O i get it. Object is from `genericmanager.SingleClusterInformerManager`, and it's unstructured type.

But can we use a typed informer here?

Reply:
This informer is shared by other parts.